libres/lib/res_util/memory.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+   Copyright (C) 2011  Equinor ASA, Norway.
+
+   The file 'memory.cpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sys/resource.h>
+
+/**
+ * NOTE: Linux only.
+ *
+ * Returns available memory (Mb) on a Linux system or -1 if something failed.
```

Comment:
Personally I try hard to avoid the use of special values like `-1`; how about `std::optional<long>` as return value?